src/daemon/custom_image_host.cpp

```diff
@@ -20,64 +20,197 @@
+    mp::VMImageInfo core_image_info{{"core"},
+                                    "Ubuntu",
+                                    "core-16",
+                                    "Core 16",
+                                    true,
+                                    image_url,
+                                    "",
+                                    "",
+                                    base_image_info.hash,
+                                    base_image_info.last_modified,
+                                    0};
+
+    default_images.push_back(core_image_info);
 
-    default_aliases.insert({"core", core_image_info});
+    auto map = map_aliases_to_vm_info_for(default_images);
 
-    return default_aliases;
+    return std::unique_ptr<mp::CustomManifest>(new mp::CustomManifest{std::move(default_images), std::move(map)});
+}
+
+auto snapcraft_default_aliases(mp::URLDownloader* url_downloader, const QString& path_prefix)
+{
+    std::vector<mp::VMImageInfo> default_images;
+    QString image_url;
+    QString hash_url;
+    QString image_file;
+
+    {
+        image_file = "ubuntu-16.04-minimal-cloudimg-amd64-disk1.img";
+        if (!path_prefix.isEmpty())
+        {
+            image_url = QUrl::fromLocalFile(path_prefix + image_file).toString();
+            hash_url = QUrl::fromLocalFile(path_prefix + "SHA256SUMS").toString();
+        }
+        else
+        {
+            const QString url_prefix{"http://cloud-images.ubuntu.com/minimal/releases/xenial/release/"};
+            image_url = url_prefix + image_file;
+            hash_url = url_prefix + "SHA256SUMS";
+        }
+
+        auto base_image_info = base_image_info_for(url_downloader, image_url, hash_url, image_file);
+        default_images.push_back(mp::VMImageInfo{{"core", "core16"},
+                                                 "",
+                                                 "snapcraft-core16",
+                                                 "Snapcraft builder for Core 16",
+                                                 true,
+                                                 image_url,
+                                                 "",
+                                                 "",
+                                                 base_image_info.hash,
+                                                 base_image_info.last_modified,
+                                                 0});
+    }
+
+    {
+        image_file = "ubuntu-18.04-minimal-cloudimg-amd64.img";
+        if (!path_prefix.isEmpty())
+        {
+            image_url = QUrl::fromLocalFile(path_prefix + image_file).toString();
+            hash_url = QUrl::fromLocalFile(path_prefix + "SHA256SUMS").toString();
+        }
+        else
+        {
+            const QString url_prefix{"http://cloud-images.ubuntu.com/minimal/releases/bionic/release/"};
+            image_url = url_prefix + image_file;
+            hash_url = url_prefix + "SHA256SUMS";
+        }
+
```

Comment:
Looks like these sections repeat with slight modifications - so maybe a method to avoid repetition?

Reply:
Yeah, I'll see what I can do about that.